javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -135,14 +133,11 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
-        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
-        Supplier<? extends T> tmp = supplier;
-        if (tmp != null) {
+        if ((value == null) && (supplier != null)) {
```

Comment:
@shipilev Thanks for the great explanation!
One question. We saw that write(value, SOME_VALUE) is synchronized with write(supplier, null) because supplier is volatile. What does another Thread see that only performs read(value). Does an execution exist such that read(value) = null (uninitialized) and supplier == null?